source/components/ch4_component.cpp

```diff
@@ -104,25 +104,22 @@ void CH4Component::setData( const string& varName,
     try {
          if( varName ==  D_PREINDUSTRIAL_CH4  ) {
             H_ASSERT( data.date == Core::undefinedIndex() , "date not allowed" );
-            M0 = unitval::parse_unitval( data.value_str, data.units_str, U_PPBV_CH4 );
```

Comment:
@rplzzz Thanks for the clarification! Now I understand differences in handling the string values in the components' `setData` method (depending on if it's an input set via `sendMessage` or not). From the API standpoint I need, however, a way to set input data without the ini-parser, that's why I call the core's `setData` method (also, it's faster if I can cache the component for setting arrays of data without using the string matches in the message system).